src/App.js

```diff
@@ -1,16 +1,64 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+
+import './App.css';
 
 const App = () => {
+  const [messagesData, setMessagesData] = useState(chatMessages);
+
+  const [numLiked, setNumLiked] = useState(0);
+
+  const onHeartClicks = (liked) => {
+    if (liked) {
+      setNumLiked(numLiked - 1);
+    } else {
+      setNumLiked(numLiked + 1);
+    }
+  };
+
+  const onUpdateLikeFx = (id) => {
+    const newMessages = messagesData.map((message) => {
+      if (message.id === id) {
+        return { ...message, liked: !message.liked };
+      } else {
+        return message;
+      }
+    });
+
+    setMessagesData(newMessages);
+  };
+
+  const userOne = messagesData[0]['sender'];
```

Comment:
What would happen if there weren't any messages yet (consider a user has opened the chat view with a friend and is getting ready to send their first message).

Prefer `.sender` to `['sender']`.